crates/storage/src/interface.rs

```diff
@@ -13,21 +13,41 @@
 #[path = "tests/interface.rs"]
 mod tests;
 
+use std::io::Error as IoError;
+use std::sync::Arc;
+
+use borsh::{to_vec, BorshDeserialize};
+use calimero_store::key::Storage as StorageKey;
+use calimero_store::layer::{ReadLayer, WriteLayer};
+use calimero_store::slice::Slice;
+use calimero_store::Store;
+use eyre::Report;
+use parking_lot::RwLock;
 use thiserror::Error as ThisError;
 
 use crate::address::{Id, Path};
 use crate::entities::Element;
 
 /// The primary interface for the storage system.
-#[derive(Debug, Default, Clone, Eq, PartialEq)]
+#[derive(Debug, Clone)]
 #[non_exhaustive]
-pub struct Interface;
+pub struct Interface {
+    /// The backing store to use for the storage interface.
+    store: Arc<RwLock<Store>>,
```

Comment:
`Store` is itself an `Arc<_>` https://github.com/calimero-network/core/blob/1db5a74d5afd88d294c10a5b758cdae383db0cbd/crates/store/src/lib.rs#L22-L24

and it doesn't need a mutual exclusion lock as none of the `Database` methods [do either](https://github.com/calimero-network/core/blob/1db5a74d5afd88d294c10a5b758cdae383db0cbd/crates/store/src/db.rs#L29-L46)

Reply:
Ahhhh yes, thanks for pointing that out! That's a nice simplification to make 😄

Reply:
Unfortunately I can't make this change, and in coming to do it I remember now trying to do this before. The reason it's in an `Arc<RwLock>` is because the `Store` demands self-mut, which I don't want to propagate onto the `Interface`.

Reply:
> demands self-mut

where?